recipes/icu/all/conanfile.py

```diff
@@ -75,6 +75,9 @@ def validate(self):
             if not os.path.exists(str(self.options.dat_package_file)):
                 raise ConanInvalidConfiguration("Non-existent dat_package_file specified")
 
+        if self.settings.compiler.get_safe("cppstd"):
```

Comment:
You'll probably also need something like [this](https://github.com/conan-io/conan-center-index/blob/7c80e9e24abaa4963b6cd42eb7dab023d4d256a2/recipes/qt/6.x.x/conanfile.py#L188-L197) and [this](https://github.com/conan-io/conan-center-index/blob/7c80e9e24abaa4963b6cd42eb7dab023d4d256a2/recipes/qt/6.x.x/conanfile.py#L258-L262). That enforces an additional check that the minimum version of the compiler supports C++17.

Reply:
Thank you for all the help! In case it wasn't obvious, this aspect of conan packages is fairly new to me.

Reply:
No problem. It's always good to see a new contributor helping out, and I'm looking forward to the new version. I should also point you to the CMake package template, [here](https://github.com/conan-io/conan-center-index/blob/master/docs/package_templates/cmake_package/all/conanfile.py), which you'll probably find really useful.